### PR TITLE
chore(weekly reports): Add more logs

### PR DESCRIPTION
**Pull request by ceorourke**
Missing weekly reports still elude me, so I'm adding more logs behind a feature flag to get more data.

### CODE REVIEW

File: src/sentry/tasks/reports.py [hunk 1]
```diff
@@ -704,6 +704,14 @@ def deliver_organization_user_report(timestamp, duration, organization_id, user_
 
     user = User.objects.get(id=user_id)
 
+    if features.has("organizations:weekly-report-debugging", organization):
+        logger.info(
+            "reports.deliver_organization_user_report.begin",
+            extra={
+                "user": user.id,
```
Comment (manuzope) [inline src/sentry/tasks/reports.py:711]: Hmmm, feels like this should be ID.

```suggestion
                "user_id": user.id,
```

**Review by manuzope — COMMENTED** [review]
Thoughts on putting something in `prepare_organization_report` to the effect of "started computing reports for x org"? That way, we'll know it ran for the org and will allow us to reduce the surface area of the investigation.

**Comment by manuzope** [comment]
I also assume you've tested this locally with settings like https://github.com/getsentry/getsentry/pull/6568?

**Review by manuzope — APPROVED** [review]

**Comment by ceorourke** [comment]
> I also assume you've tested this locally with settings like [getsentry/getsentry#6568](https://github.com/getsentry/getsentry/pull/6568)?

@manuzope I've been trying to test this locally and haven't gotten anywhere, I'm not sure if this is just something you can't test locally or I'm not doing something right.

I deployed the getsentry PR and changed my local org slug to one in the list, ran the task manually and it fails to find my organization in the feature list. I've also ran the `test_deliver_organization_user_report` test and added in the test's org slug which didn't work. The only way I can get any print statements to run (I added prints just before the logs that are feature flagged) is by enabling the feature in `server.py`.

I read through the [feature flag docs](https://develop.sentry.dev/feature-flags/#add-your-feature-to-the-featuremanager ) and removed the `True` parameter as suggested if not using Flagr but that didn't change anything for my local testing either, but I've committed that since it seems like what I'm supposed to do. 

At this point (2 on Friday) I might just deploy this PR soon if I haven't heard back from anyone on how to test this locally and hope for the best on Monday, because I am out of ideas.

**Pull request merged**